graphdatascience/graph/graph_result.py

```diff
@@ -0,0 +1,35 @@
+from dataclasses import dataclass
+from types import TracebackType
+from typing import Optional, Type, TypeVar
+
+from ..graph.graph_object import Graph
+
+TGraphResult = TypeVar("TGraphResult", bound="GraphResult")
+
+
+@dataclass(frozen=True)
+class GraphResult:
```

Comment:
In order to not be a breaking change, this class needs to operate just like a Tuple. I would've expected it to inherit from `Tuple`, but perhaps you've managed to just implement the methods a Tuple requires?

Reply:
The name here is interesting. I want to think back to our session on "conceptualising graph projection" at GDSCon Spring 2023, where we defined the operation of "graph creation". I think this type will always be returned when any of the "graph creation" operations is used; projections, generation, import, construct, etc. (`graph.get` is a bit of an exception IMO; it doesn't really _do_ anything).

So I would like to consider these names

```
GraphCreationResult
GraphCreateResult
GraphCreation
```

My preference is `GraphCreationResult`. WDYT?

Reply:
I like the GraphCreationResult. Thanks for the naming ideas